src/wallet/rpcwallet.cpp

```diff
@@ -3241,43 +3292,82 @@ static UniValue bumpfee(const JSONRPCRequest& request)
                     HelpExampleCli("bumpfee", "<txid>")
                 },
             }.ToString());
+        }
     }
 
     RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VOBJ});
     uint256 hash(ParseHashV(request.params[0], "txid"));
 
     // optional parameters
-    CAmount totalFee = 0;
+    CAmount total_fee = 0;
     CCoinControl coin_control;
     coin_control.m_signal_bip125_rbf = true;
-    if (!request.params[1].isNull()) {
-        UniValue options = request.params[1];
-        RPCTypeCheckObj(options,
-            {
-                {"confTarget", UniValueType(UniValue::VNUM)},
-                {"totalFee", UniValueType(UniValue::VNUM)},
-                {"replaceable", UniValueType(UniValue::VBOOL)},
-                {"estimate_mode", UniValueType(UniValue::VSTR)},
-            },
-            true, true);
 
-        if (options.exists("confTarget") && options.exists("totalFee")) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, "confTarget and totalFee options should not both be set. Please provide either a confirmation target for fee estimation or an explicit total fee for the transaction.");
-        } else if (options.exists("confTarget")) { // TODO: alias this to conf_target
-            coin_control.m_confirm_target = ParseConfirmTarget(options["confTarget"]);
-        } else if (options.exists("totalFee")) {
-            totalFee = options["totalFee"].get_int64();
-            if (totalFee <= 0) {
-                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid totalFee %s (must be greater than 0)", FormatMoney(totalFee)));
+    if (!IsDeprecatedRPCEnabled("bumpfee")) {
+        if (!request.params[1].isNull()) {
+            UniValue options = request.params[1];
+            RPCTypeCheckObj(options,
+                {
+                    {"confTarget", UniValueType(UniValue::VNUM)},
+                    {"total_fee", UniValueType(UniValue::VNUM)},
+                    {"replaceable", UniValueType(UniValue::VBOOL)},
+                    {"estimate_mode", UniValueType(UniValue::VSTR)},
+                },
+                true, true);
+
+            if (options.exists("confTarget") && options.exists("total_fee")) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "confTarget and total_fee options should not both be set. Please provide either a confirmation target for fee estimation or an explicit total fee for the transaction.");
+            } else if (options.exists("confTarget")) { // TODO: alias this to conf_target
+                coin_control.m_confirm_target = ParseConfirmTarget(options["confTarget"]);
+            } else if (options.exists("total_fee")) {
+                total_fee = AmountFromValue(options["total_fee"]); // Convert to satoshis
+                if (total_fee >= 100000000) { // User entered satoshis
+                    throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid total_fee, enter a value in bitcoins.");
```

Comment:
The error should include why the `total_fee` is invalid (as done below).

Also this error condition should be tested :-)